src/main/java/seedu/address/logic/commands/consultation/ImportConsultCommand.java

```diff
@@ -0,0 +1,239 @@
+                    }
+                } catch (IllegalArgumentException e) {
+                    errorEntries.add(new String[]{line, e.getMessage()});
+                    errorCount++;
+                    logger.fine("Error parsing consultation entry: " + e.getMessage());
+                }
+            }
+
+            if (!hasValidEntries) {
+                throw new CommandException(MESSAGE_EMPTY_FILE);
+            }
+
+            if (!errorEntries.isEmpty()) {
+                Path errorPath = writeErrorFile(errorEntries);
+                return new CommandResult(
+                        String.format(MESSAGE_SUCCESS_WITH_ERRORS, successCount, errorCount, errorPath),
+                        COMMAND_TYPE);
+            }
+
+            return new CommandResult(String.format(MESSAGE_SUCCESS, successCount, errorCount), COMMAND_TYPE);
+
+        } catch (IOException e) {
+            logger.warning("Error reading import file: " + e.getMessage());
+            throw new CommandException(String.format(MESSAGE_INVALID_FILE, e.getMessage()));
+        }
+    }
+
+    /**
+     * Resolves the file path, handling both home directory paths (starting with ~) and parent directory paths.
+     */
+    protected Path resolveFilePath(String filepath) {
+        if (filepath.startsWith("~")) {
+            return Paths.get(System.getProperty("user.home"))
+                    .resolve(filepath.substring(2));
+        }
+
+        if (!Paths.get(filepath).isAbsolute()) {
+            return Paths.get(System.getProperty("user.dir"))
+                    .getParent()
+                    .resolve(filepath);
+        }
+
+        return Paths.get(filepath);
+    }
+
+    private Path writeErrorFile(List<String[]> errorEntries) throws IOException {
+        Path errorPath = filePath.resolveSibling("error.csv");
+        try (FileWriter writer = new FileWriter(errorPath.toFile())) {
+            writer.write("Original Entry,Error Message\n");
+            for (String[] entry : errorEntries) {
+                writer.write(String.format("%s,%s\n",
+                        escapeSpecialCharacters(entry[0]),
+                        escapeSpecialCharacters(entry[1])));
+            }
+        }
+        return errorPath;
+    }
+
+    /**
+     * Parses a CSV line into a Consultation object.
+     */
+    protected Consultation parseConsultation(String line, Model model) throws IllegalArgumentException {
+        String[] parts = line.split(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)", -1);
+        if (parts.length < 3) {
+            throw new IllegalArgumentException("Incomplete consultation entry");
+        }
+
+        try {
+            String date = parts[0].trim();
+            String time = parts[1].trim();
+            String studentsStr = unescapeSpecialCharacters(parts[2]);
+
+            List<Student> students = new ArrayList<>();
+            if (!studentsStr.isEmpty()) {
```

Comment:
This potentially falls into arrowhead code territory. Maybe consider extracting the code into another method?